src/graph.rs

```diff
@@ -1195,20 +1223,55 @@ mod tests {
         graph.insert(Edge::new(2, 6));
         graph.insert(Edge::new(1, 3));
 
-        let between_map = graph.betweenness_centrality(1);
+        // passing in zero as num_threads will be clamped to 1 thread
+        let between_map = graph.betweenness_centrality(0, false);
+        let close_map = graph.closeness_centrality(0);
+        const N: usize = 7;
+        let mut betweenness = [0.0; N];
+        let mut closeness = [0.0; N];
+        for i in 0..N {
+            betweenness[i] = *between_map.get(&i).unwrap();
+            closeness[i] = *close_map.get(&i).unwrap();
+        }
+
+        let total_path_length = [15, 9, 10, 12, 15, 12, 15];
+        let mut expected_closeness = [0.0; N];
+        let expected_betweenness = [0.5, 9.5, 9.0, 2.0, 0.0, 2.0, 0.0];
+        for i in 0..N {
+            expected_closeness[i] = total_path_length[i] as f64 / (N - 1) as f64;
+        }
+
+        assert_eq!(betweenness, expected_betweenness);
+        assert_eq!(closeness, expected_closeness);
+    }
+
+    #[test]
+    fn randomish_graph_normalized() {
+        let mut graph: Graph<usize> = Graph::new();
+        graph.insert(Edge::new(0, 3));
+        graph.insert(Edge::new(0, 5));
+        graph.insert(Edge::new(5, 1));
+        graph.insert(Edge::new(1, 2));
+        graph.insert(Edge::new(2, 4));
+        graph.insert(Edge::new(2, 6));
+        graph.insert(Edge::new(1, 3));
+
+        const N: usize = 7;
+        let between_map = graph.betweenness_centrality(1, true);
         let close_map = graph.closeness_centrality(1);
-        let mut betweenness: [f64; 7] = [0.0; 7];
-        let mut closeness: [f64; 7] = [0.0; 7];
-        for i in 0..7 {
+        let mut betweenness = [0.0; N];
+        let mut closeness = [0.0; N];
+        for i in 0..N {
             betweenness[i] = *between_map.get(&i).unwrap();
             closeness[i] = *close_map.get(&i).unwrap();
         }
 
         let total_path_length = [15, 9, 10, 12, 15, 12, 15];
-        let mut expected_closeness: [f64; 7] = [0.0; 7];
-        let expected_betweenness: [f64; 7] = [0.5, 9.5, 9.0, 2.0, 0.0, 2.0, 0.0];
-        for i in 0..7 {
-            expected_closeness[i] = total_path_length[i] as f64 / 6.0;
+        let mut expected_closeness = [0.0; N];
+        let mut expected_betweenness = [0.5, 9.5, 9.0, 2.0, 0.0, 2.0, 0.0];
+        for i in 0..N {
+            expected_closeness[i] = total_path_length[i] as f64 / (N - 1) as f64;
+            expected_betweenness[i] /= ((N - 1) * (N - 2) / 2) as f64;
```

Comment:
The same question like for DIVISOR const a little above 😃

Reply:
I just checked in a change, so this second test case with normalization uses the DIVISOR same as in the first case. 😄